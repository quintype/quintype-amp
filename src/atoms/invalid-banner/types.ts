import { Story } from "../../types/story";
import { DefaultTheme, CSSObject } from "styled-components";

export interface InvalidBannerTypes {
  story: Story;
  style?: CustomStyles;
  theme?: DefaultTheme;
}

interface CustomStyles {
  wrapper?: (theme) => CSSObject;
  banner?: (theme) => CSSObject;
}
