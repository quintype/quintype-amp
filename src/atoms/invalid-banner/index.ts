import { InvalidBanner } from "./invalid-banner";
export { InvalidBanner };
