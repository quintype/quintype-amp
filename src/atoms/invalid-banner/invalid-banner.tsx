import React from "react";
import styled from "styled-components";
import { withStory } from "../../context";

const StyledBanner = styled.div`
  width: 100%;
  text-align: center;
  position: absolute;
  top: 60px;
  line-height: 50px;
  background-color: ${(props) => props.theme.color.mono4};
`;

const InvalidBannerBase = ({ story }) => {
  const storyUrl = story.url;
  return (
    <StyledBanner>
      Some of the elements in this story are not compatible with AMP. To view the complete story, please{" "}
      <a href={storyUrl}>click here</a>
    </StyledBanner>
  );
};

export const InvalidBanner = withStory(InvalidBannerBase);
