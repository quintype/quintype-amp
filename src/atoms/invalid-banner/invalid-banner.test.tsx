import React from "react";
import { InvalidBanner } from "./invalid-banner";
import { shallow } from "enzyme";

describe("InvalidBanner", () => {
  it("should render", () => {
    const wrapper = shallow(<InvalidBanner />);
    expect(wrapper).toMatchSnapshot();
  });
});
