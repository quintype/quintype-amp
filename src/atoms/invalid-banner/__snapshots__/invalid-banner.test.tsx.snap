// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvalidBanner should render 1`] = `
<InvalidBannerBase
  story={Object {}}
/>
`;
